### PR TITLE
MTDSA-6814 Employments endpoints alignments

### PR DESCRIPTION
**Pull request by Taiwo-AB**

### CODE REVIEW

File: app/v1/controllers/requestParsers/validators/validations/EmployerNameValidation.scala [hunk 1]
```diff
@@ -20,6 +20,6 @@ import v1.models.errors.{EmployerNameFormatError, MtdError}
 
 object EmployerNameValidation {
   def validate(employerName: String, maxLength: Int): List[MtdError] = {
-    if(employerName.length() <= maxLength) NoValidationErrors else List(EmployerNameFormatError)
+    if(employerName.length >= 1 && employerName.length <= maxLength) NoValidationErrors else List(EmployerNameFormatError)
```
Comment (yangwong9664) [inline app/v1/controllers/requestParsers/validators/validations/EmployerNameValidation.scala:23]: `AmendOtherEmployment` uses this validation too, even though the only changes needed are supposed to be on `AddCustomEmployment` and `AmendCustomEmployment`. Have you checked `AmendOtherEmployment` needs the `>=1` update?

File: app/v1/controllers/requestParsers/validators/validations/PayrollIdValidation.scala [hunk 2]
```diff
@@ -26,7 +26,7 @@ object PayrollIdValidation {
   }
 
   def validate(payrollId: String): List[MtdError] = {
```
Comment (yangwong9664) [inline app/v1/controllers/requestParsers/validators/validations/PayrollIdValidation.scala:28]: This method can be private as no outside classes call it directly, every request goes through `validateOptional`.

File: resources/public/api/conf/1.0/schemas/employments_definitions.json [hunk 3]
```diff
@@ -40,8 +41,8 @@
     },
     "submittedOn": {
       "type": "string",
-      "description": "The date the custom employment was added. It must be provided in the format YYYY-MM-DD",
-      "example": "2020-07-06"
+      "description": "The date the custom employment was added. It must be provided in the format YYYY-MM-DDThh:mm:ss.ssZ",
```
Comment (yangwong9664) [inline resources/public/api/conf/1.0/schemas/employments_definitions.json:44]: `YYYY-MM-DDThh:mm:ss:ssZ`

**Review by yangwong9664 — CHANGES_REQUESTED** [review]

File: app/v1/controllers/requestParsers/validators/validations/EmployerNameValidation.scala (hunk 1 above)
Comment (Taiwo-AB) [reply app/v1/controllers/requestParsers/validators/validations/EmployerNameValidation.scala:23]: of course I did

File: app/v1/controllers/requestParsers/validators/validations/PayrollIdValidation.scala (hunk 2 above)
Comment (Taiwo-AB) [reply app/v1/controllers/requestParsers/validators/validations/PayrollIdValidation.scala:28]: I was only asked to change the regex for this. If this is the case we will need to go through the whole codebase to make this sort of changes.

**Pull request merged**